azurerm/internal/services/datafactory/data_factory_linked_service_synapse_resource.go

```diff
@@ -0,0 +1,289 @@
+				DiffSuppressFunc: azureRmDataFactoryLinkedServiceConnectionStringDiff,
+				ValidateFunc:     validation.StringIsNotEmpty,
+			},
+
+			"key_vault_password_reference": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"linked_service_name": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+
+						"password_secret_name": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+					},
+				},
+			},
+
+			"description": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"integration_runtime_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"parameters": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"annotations": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"additional_properties": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func resourceArmDataFactoryLinkedServiceSynapseCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).DataFactory.LinkedServiceClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	dataFactoryName := d.Get("data_factory_name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	if d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroup, dataFactoryName, name, "")
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of existing Data Factory Linked Service Synapse%q (Data Factory %q / Resource Group %q): %+v", name, dataFactoryName, resourceGroup, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_data_factory_linked_service_synapse", *existing.ID)
+		}
+	}
+
+	passwordReference := d.Get("key_vault_password_reference").([]interface{})
+
+	sqlDWProperties := &datafactory.AzureSQLDWLinkedServiceTypeProperties{
+		ConnectionString: d.Get("connection_string").(string),
+		Password:         expandAzureKeyVaultPasswordReference(passwordReference),
+	}
+
+	description := d.Get("description").(string)
+
+	sqlDWLinkedService := &datafactory.AzureSQLDWLinkedService{
+		Description:                           &description,
+		AzureSQLDWLinkedServiceTypeProperties: sqlDWProperties,
+		Type:                                  datafactory.TypeAzureSQLDW,
+	}
```

Comment:
could we nest these declarations? 
```suggestion
	passwordReference := d.Get("key_vault_password_reference").([]interface{})
	
	sqlDWLinkedService := &datafactory.AzureSQLDWLinkedService{
		Description:                           utils.String(d.Get("description").(string)),
		AzureSQLDWLinkedServiceTypeProperties: &datafactory.AzureSQLDWLinkedServiceTypeProperties{
			ConnectionString: d.Get("connection_string").(string),
			Password:         expandAzureKeyVaultPasswordReference(passwordReference),
		},
		Type:                                  datafactory.TypeAzureSQLDW,
	}
```